x-pack/plugins/aiops/public/components/log_categorization/format_category.tsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useMemo } from 'react';
+import { EuiText, EuiHorizontalRule } from '@elastic/eui';
+import { css, SerializedStyles } from '@emotion/react';
+import type { Category } from '../../../common/api/log_categorization/types';
+import { useIsDarkTheme } from '../../hooks/use_eui_theme';
+
+interface Props {
+  category: Category;
+  count?: number;
+}
+
+// copies of the syntax colors used in EuiCode
+const tokenStyleLight = css`
+  color: #765b96;
+`;
+const wildcardStyleLight = css`
+  color: #357160;
+`;
+const tokenStyleDark = css`
+  color: #a68ac5;
+`;
+const wildcardStyleDark = css`
+  color: #54b399;
+`;
+
+interface Styles {
+  tokenStyle: SerializedStyles;
+  wildcardStyle: SerializedStyles;
+}
+
+const useStyles = (): Styles => {
+  const isDarkTheme = useIsDarkTheme();
+  return isDarkTheme
+    ? {
+        tokenStyle: tokenStyleDark,
+        wildcardStyle: wildcardStyleDark,
+      }
+    : {
+        tokenStyle: tokenStyleLight,
+        wildcardStyle: wildcardStyleLight,
+      };
+};
+
+export const useCreateFormattedExample = () => {
+  const { tokenStyle, wildcardStyle } = useStyles();
+
+  function createFormattedExample(key: string, example: string): JSX.Element[] {
```

Comment:
Nit: This function could be wrapped in `useCallback()` to avoid returning a new instance every time. Otherwise I think the `useMemo` in `FormattedPatternExamples` might not memoize as intended.

Reply:
Updated in https://github.com/elastic/kibana/pull/175320/commits/c45c84906cc2524aa0e9d1d2863c65a91fb8130c